internal/services/pam/pam.go

```diff
@@ -52,24 +55,46 @@ func (s Service) AvailableBrokers(ctx context.Context, _ *authd.Empty) (*authd.A
 }
 
 // GetPreviousBroker returns the previous broker set for a given user, if any.
+// If the user is not in our cache, it will try to check if it’s on the system, and return then "local".
 func (s Service) GetPreviousBroker(ctx context.Context, req *authd.GPBRequest) (*authd.GPBResponse, error) {
+	// Use in memory cache first
 	if b := s.brokerManager.BrokerForUser(req.GetUsername()); b != nil {
 		return &authd.GPBResponse{PreviousBroker: b.ID}, nil
 	}
 
+	// Load from database cache.
 	brokerID, err := s.userManager.BrokerForUser(req.GetUsername())
-	if err != nil {
+	// User is not in our cache.
+	if err != nil && errors.Is(err, cache.NoDataFoundError{}) {
```

Comment:
It looks like this is the only place in which we need the `users/cache` package for, so it should be easy to address the previous comment.